### PR TITLE
Magento2 685 phi product category assignments

### PR DESCRIPTION
**Pull request by andre-kraus**
# Pull Request Template

## Description

Categories were missing in the items export.

## Type of change

- [X] Bug fix
- [ ] New feature
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I updated the CHANGELOG.md

### CODE REVIEW

**Review by augsteyer — CHANGES_REQUESTED** [review]
tested, works (simple, config, grouped), but we will need tests with it too. Patch the merchant.

Also an issue I am seeing with configurable and grouped products, I am seeing many blank lines in nodes:
```
                    <identifier type="SKU">
                        <![CDATA[MH01-XS-Black]]>
                    
                    
                    </identifier>
```

**Comment by augsteyer** [comment]
Nevermind the blank lines, might have been a glitch with postman

File: src/Helper/Product/Utility.php [hunk 1]
```diff
@@ -140,7 +140,7 @@ public function isVisibleInCategories($product): bool
             Visibility::VISIBILITY_IN_CATALOG
         ];
 
-        return in_array($product->getVisibility(), $validVisibilities, true);
+        return in_array($product->getVisibility(), $validVisibilities, false);
```
Comment (Skistims) [inline src/Helper/Product/Utility.php]: According to our coding standards, I think we should cast the type and then have the type sensitive comparison.
At least we did it this way in the past

**Review by Skistims — CHANGES_REQUESTED** [review]

File: src/Helper/Product/Utility.php (hunk 1 above)
Comment (augsteyer) [reply src/Helper/Product/Utility.php]: kept the strict comparison

**Comment by augsteyer** [comment]
> tested, works (simple, config, grouped), but we will need tests with it too. Patch the merchant.
> 
> Also an issue I am seeing with configurable and grouped products, I am seeing many blank lines in nodes:
> 
> ```
>                     <identifier type="SKU">
>                         <![CDATA[MH01-XS-Black]]>
>                     
>                     
>                     </identifier>
> ```

There were existing tests, but they did not cover the type issue. Some tests were created and some moved out to a different class.

File: src/Helper/Product/Utility.php [hunk 2]
```diff
@@ -49,6 +49,7 @@
 use Shopgate_Model_Catalog_Relation;
 use Shopgate_Model_Catalog_Validation;
 use Shopgate_Model_Catalog_Visibility;
+use function in_array;
```
Comment (Skistims) [inline src/Helper/Product/Utility.php:52]: can you explain what this is for?
I looked into the usage of php core functions this way ind it seems to make sense if you want to use an alias.
This seems rather useless

Comment (augsteyer) [reply src/Helper/Product/Utility.php:52]: this is a performance optimization tweak based on: https://github.com/Roave/FunctionFQNReplacer

Comment (Skistims) [reply src/Helper/Product/Utility.php:52]: I thought readability is the highes priority for our projects.
This is a line of code, which does not change the outcome and is thereby an abstraction.

But if you like it, keep it.

**Review by Skistims — APPROVED** [review]

File: src/Helper/Product/Utility.php (hunk 2 above)
Comment (augsteyer) [reply src/Helper/Product/Utility.php:52]: Hm? Performance optimization is the outcome imho, readability of the use case block is sort of meh... Unless we disagree and then we may need a team meeting on it ^_^

**Review by augsteyer — APPROVED** [review]

**Comment by augsteyer** [comment]
gr, I had to commit tests for attributes + some refactoring. Need someone to take a looksie at it again.

@andre-kraus can you merge this and release a version? I believe one of the merchants needed a composer version.

**Pull request merged**